scripts/config/msu/classes/enum.nut

```diff
@@ -0,0 +1,49 @@
+::MSU.Class.Enum <- class
+{
+	_Keys = null;
+	_ValueKeyArray = null;
+
+	constructor( _array = null )
+	{
+		this._Keys = {};
+		this._ValueKeyArray = [];
```

Comment:
Might be more performant to have this as a table too with integers as its keys. Because `push` will normally resize the array to double the size (aka more memory is reserved) - not sure if this is how squirrel works but some programming languages do.

Reply:
tables are ~3x more memory intensive than arrays in squirrel according to electricimp

Reply:
Wow, that is interesting information. I suppose this should be kept in mind - because in certain places we are using tables to enhance "performance" but as BB is primarily a memory limited game, we should be careful about this. @TaroEld